lib/getTeam.ts

```diff
@@ -0,0 +1,66 @@
+import { Team } from "@prisma/client";
+import { GetServerSidePropsContext } from "next";
+import prisma from "./prisma";
+
+export const getTeam = async (context: GetServerSidePropsContext): Promise<Team | null> => {
+  let teamIdOrSlug = null;
+  let team = null;
+
+  switch (process.env.NODE_ENV) {
+    case "development": {
+      if (
+        process.env?.TESTING_TEAMS_CUSTOM_DOMAINS &&
+        Boolean(parseInt(process.env.TESTING_TEAMS_CUSTOM_DOMAINS))
+      ) {
+        teamIdOrSlug = process.env.TESTING_TEAMS_TEAM ?? "1";
+      }
+      break;
+    }
+
+    case "production":
+    default: {
+      const host = context.req.headers.host;
+      teamIdOrSlug = host.split(".")[0];
+      break;
+    }
+  }
+
+  const teamSelectInput = {
+    id: true,
+    name: true,
+    slug: true,
+    members: {
+      select: {
+        user: {
+          select: {
+            id: true,
+            username: true,
+            email: true,
+            name: true,
+            bio: true,
+            avatar: true,
+            theme: true,
+          },
+        },
+      },
+    },
+  };
+
+  if (teamIdOrSlug && parseInt(teamIdOrSlug)) {
+    team = await prisma.team.findFirst({
```

Comment:
```javascript
whereAndSelect(
  prisma.team.findFirst, 
  {
    id: parseInt(teamIdOrSlug)
  }, [
    "id", "name", "slug",
    "members.user.id", "members.user.username", "members.user.email", "members.user.name",
    "members.user.bio", "members.user.avatar", "members.user.theme"
  ]
);
```

Reply:
Additionally, have a look at removing the if clauses to determine whether it is a slug or an id altogether, and go for a more robust `OR` type of approach.

I think `$where: { id: parseInt(teamIdOrSlug), slug: teamIdOrSlug }` may already yield success  selecting by either id or slug. If that does not work, have a look at doing that using OR. I've made this work before so if you need me to find out, just let me know.